### ISSUE TITLE
Lambda layers research

### ISSUE THREAD

**Issue by steebchen** [prisma/specs#379]
Photon requires users to know about our rust binaries. When deploying to Serverless environments, such as AWS Lambda, they need to know in advance which binary is required, and specify that binary in the `schema.prisma` file. Photon Go can fetch the binary automatically as per https://github.com/prisma/photongo/pull/61, but this results in slow cold boots (warm boots are fast in any case).

Lambda layers are pre-defined zip files with dependencies which serverless functions can build on top of. This means we can pre-build Lambda layers, save them on S3, and our users can just reference these layers so the binary is already fetched, even on cold starts.

However, this has one major caveat, which is versioning. We already have builtin versioning in PhotonJS and Photon Go, and Lambda layers break versioning, or we need to come up with a new or “third party” versioning system. Since versioning with a third party tool would mean it’s not self-contained anymore, it’s unlikely we’ll provide or recommend using layers for now.

TLDR: While the concept of Lambda layers is interesting, I don't think we should make use of it.
